pydicom_seg/reader.py

```diff
@@ -167,8 +167,14 @@ def read(self, dataset: pydicom.Dataset) -> SegmentReadResult:
 
             # Iterate over all frames and check for referenced segment number
             for frame_idx, pffg in enumerate(dataset.PerFrameFunctionalGroupsSequence):
-                if segment_number != pffg.SegmentIdentificationSequence[0].ReferencedSegmentNumber:
-                    continue
+                if hasattr(pffg, 'SegmentIdentificationSequence'):
+                    sis = pffg.SegmentIdentificationSequence[0]
```

Comment:
```python
sis = pffg.get('SegmentIdentificationSequence', shared_sis)
```

Reply:
Your solution is better so I think that is the one to keep. I made a quick test and your changes just need  'if segment_number != sis[0].ReferencedSegmentNumber:' instead of if segment_number != sis.ReferencedSegmentNumber: